bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/A5_09/A5_09_04.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.enocean.internal.eep.A5_09;
+
+import static org.openhab.binding.enocean.internal.EnOceanBindingConstants.*;
+
+import java.util.function.Function;
+
+import org.openhab.binding.enocean.internal.messages.ERP1Message;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+
+/**
+ *
+ * @author Zhivka Dimova
+ */
+public class A5_09_04 extends A5_09 {
+
+    public A5_09_04() {
+        super();
+    }
+
+    public A5_09_04(ERP1Message packet) {
+        super(packet);
+    }
+
+    protected double getUnscaledCO2Min() {
+        return 0;
+    }
+
+    protected double getUnscaledCO2Max() {
+        return 255;
+    }
+
+    protected double getScaledCO2Min() {
+        return 0;
+    }
+
+    protected double getScaledCO2Max() {
+        return 2550;
+    }
+
+    protected int getUnscaledCO2Value() {
+        return getDB_2Value();
+    }
+
+    protected double getUnscaledTemperatureMin() {
+        return 0;
+    }
+
+    protected double getUnscaledTemperatureMax() {
+        return 255;
+    }
+
+    protected double getScaledTemperatureMin() {
+        return 0;
+    }
+
+    protected double getScaledTemperatureMax() {
+        return 51;
+    }
+
+    protected int getUnscaledTemperatureValue() {
+        return getDB_1Value();
+    }
+
+    protected double getUnscaledHumidityMin() {
+        return 0;
+    }
+
+    protected double getUnscaledHumidityMax() {
+        return 200;
+    }
+
+    protected int getUnscaledHumidityValue() {
+        return getDB_3Value();
+    }
+
+    protected double getScaledHumidityMin() {
+        return 0;
+    }
+
+    protected double getScaledHumidityMax() {
+        return 100;
+    }
+
+    @Override
+    protected State convertToStateImpl(String channelId, String channelTypeId,
+            Function<String, State> getCurrentStateFunc, Configuration config) {
+
+        if (channelId.equals(CHANNEL_CO2)) {
+            return new DecimalType(
+                    getScaledCO2Min() + ((getUnscaledCO2Value() * (getScaledCO2Max() - getScaledCO2Min()))
+                            / (getUnscaledCO2Max() - getUnscaledCO2Min())));
```

Comment:
For UoM, something like:
```suggestion
            return new QuantityType<>(
                    getScaledCO2Min() + ((getUnscaledCO2Value() * (getScaledCO2Max() - getScaledCO2Min()))
                            / (getUnscaledCO2Max() - getUnscaledCO2Min())), Units.PARTS_PER_MILLION);
```